execution_engine_testing/tests/src/test/contract_api/get_call_stack.rs

```diff
@@ -2620,7 +2620,10 @@ mod payment {
         } else {
             builder.exec(execute_request).commit().expect_failure();
             let error = builder.get_error().expect("must have an error");
-            assert!(matches!(error, CoreError::Exec(ExecError::GasLimit)));
+            assert!(matches!(
+                error,
+                CoreError::Exec(ExecError::InvalidContext) | CoreError::Exec(ExecError::GasLimit)
```

Comment:
It's not clear to me why we'd sometimes get an `InvalidContext` error now.

Maybe we could extract the whole `if call_depth == 0` block to a separate function and a comment there.  It's repeated 5 times in this mod I think.

Reply:
Refactored this duplicated assertion,

So before increasing control flow codes it was meant to be failing at InvalidContext, then in 1.4.9 this test was heavily refactored to always assume GasLimit (which was incorrect; but it was passing due to high costs). By lowering all control flow opcodes except br, br_table, br_if I had to modify this again.